driver/src/main/java/org/neo4j/driver/internal/net/pooling/PoolSettings.java

```diff
@@ -53,4 +57,19 @@ public boolean idleTimeBeforeConnectionTestConfigured()
     {
         return idleTimeBeforeConnectionTest >= 0;
     }
+
+    public long maxConnectionLifetime()
+    {
+        if ( !maxConnectionLifetimeConfigured() )
+        {
+            throw new IllegalStateException(
+                    "Max connection lifetime is not configured: " + maxConnectionLifetime );
+        }
+        return maxConnectionLifetime;
+    }
+
+    public boolean maxConnectionLifetimeConfigured()
```

Comment:
It might be okay to assume -1 or negative indicate `infinite` or `forever`, rather than not configured.